packages/react-storage/src/components/StorageBrowser/context/controls/Navigate/Navigate.tsx

```diff
@@ -40,13 +27,73 @@ export function navigateReducer(
   state: NavigateState,
   action: NavigateAction
 ): NavigateState {
-  if (action.type === 'SELECT_LOCATION') {
-    return {
-      ...state,
-      location: { ...state.location, current: action.location },
-    };
+  switch (action.type) {
+    case 'SELECT_LOCATION': {
+      /** 
+        This action comes from the navigate item specifying the location
+        It uses the current state location and so this has the "old" scope
+        which could include the previous folder we were at
+        We'll need to update the scope here
+        We need to update the prefix too?
+      */
```

Comment:
Nit: prefer to only use either JSDoc syntax for comments that provide information to consumers or `//` comments to explain behaviors